spec/unit/cds_synchronizer_spec.rb

```diff
@@ -0,0 +1,114 @@
+RSpec.describe CdsSynchronizer do
+  describe '.downloaded_todays_file?' do
+    before do
+      allow(TariffSynchronizer::CdsUpdate).to receive(:downloaded_todays_file?).and_return(true)
+    end
+
+    it 'calls through to CdsUpdate#downloaded_todays_file?' do
+      described_class.downloaded_todays_file?
+
+      expect(TariffSynchronizer::CdsUpdate).to have_received(:downloaded_todays_file?)
+    end
+  end
+
+  describe '.apply' do
+    let(:applied_update) { create(:cds_update, :applied, example_date: Time.zone.yesterday) }
+    let(:pending_update) { create(:cds_update, :pending, example_date: Time.zone.today) }
+
+    context 'when the sequence of updates is not correct' do
+      before do
+        applied_date = Time.zone.yesterday
+        pending_date = applied_date + 2.days
+
+        create :cds_update, :applied, example_date: applied_date, filename: "tariff_dailyExtract_v1_#{applied_date.strftime('%Y%m%d')}T123456.gzip"
+
+        create :cds_update, example_date: pending_date, filename: "tariff_dailyExtract_v1_#{pending_date.strftime('%Y%m%d')}T123456.gzip"
+      end
+
+      # rubocop:disable RSpec/MultipleExpectations
+      it 'raises wrong sequence error and notifies Slack app' do
+        allow(SlackNotifierService).to receive(:new).and_call_original
+
+        expect { described_class.apply }.to raise_error(BaseSynchronizer::FailedUpdatesError)
+
+        expect(SlackNotifierService).to have_received(:new)
```

Comment:
It is not `call` a class method?
In other parts, we test it this way:

```
  allow(SlackNotifierService).to receive(:call)
   ...
  expect(SlackNotifierService).to have_received(:call)
```

Reply:
I think its not anymore for consistency we always instantiate a stateless service. I mainly wanted to know it gets called and we test the call method in isolation elsewhere